static/src/javascripts/projects/common/modules/user-prefs.spec.js

```diff
@@ -0,0 +1,112 @@
+// @flow
+import userPrefs from 'projects/common/modules/user-prefs';
+import { local } from 'lib/storage';
+
+class LocalStorageMock {
+    store: Object;
+
+    constructor() {
+        this.store = {};
+    }
+
+    clear() {
+        this.store = {};
+    }
+
+    getItem(key) {
+        return this.store[key];
+    }
+
+    setItem(key, value) {
+        this.store[key] = value.toString();
+    }
+
+    removeItem(key) {
+        this.store[key] = null;
+    }
+}
+
+const localStorage = new LocalStorageMock();
+
+describe('userPrefs - Client-side preferences', () => {
+    beforeAll(() => {
+        local.storage = localStorage;
```

Comment:
I prefer stubs to using mocks when you can completely mimic the interface with an alternative implementation, but a manual mock sounds like a good idea. I wasn't aware of them until @desbo pointed me in their direction yesterday.